### ISSUE TITLE
Docker task extracts 'image.base.ref.name' from Dockerfile wrongly when 'from' string is found 

### ISSUE THREAD

**Issue by jonathansp** [microsoft/azure-pipelines-tasks#15117]
**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: 
* Docker (docker-common-v2)

## Environment
- Server - Azure Pipelines or TFS on-premises?
    - If using TFS on-premises, provide the version: N/A
    - If using Azure Pipelines, provide the account name, team project name, build definition name/build number: N/A
- Agent - Hosted or Private: 
    - If using Hosted agent, provide agent queue name: N/A
    - If using private agent, provide the OS of the machine running the agent and the agent version:  N/A
## Issue Description

Giving the following task definition:

```yaml
  - task: Docker@2
    displayName: Build docker image
    inputs:
      tags: |
        latest
      command: build
      buildContext: $(Build.Repository.LocalPath)
```
And the following Dockerfile
```
FROM python:3.9.6-slim
COPY . /app
RUN ./app/run from-csv=/path/to/file.csv
```
Pipeline output command will be 
```
/usr/bin/docker build
    -f /agent/_work/5/s/Dockerfile
     (...) # omited to save space
    --label image.base.ref.name=-csv=/path/to/file.csv.  <--- wrong 
```
causing error in the execution, as the `image.base.ref.name` was no parsed correctly. The bug is caused due to a misinterpretation of the function [getBaseImageName(dockerFileContent: string)](https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/docker-common-v2/containerimageutils.ts#L42), that founds the string **_from_** in the RUN command, and parses it as a docker image name.

This can be easily reproduced with this snippet (copied from the package and removed typescript syntax): 

```javascript
function getBaseImageName(dockerFileContent) {
    if (!dockerFileContent || dockerFileContent == "") {
        return null;
    }

    var lines = dockerFileContent.split(/[\r?\n]/);
    var aliasToImageNameMapping = {}
    var baseImage = "";

    for (var i = 0; i < lines.length; i++) {
        var index = lines[i].toUpperCase().indexOf("FROM");
        if (index == -1) {
            continue;
        }

        var nameComponents = lines[i].substring(index + 4).toLowerCase().split(" as ");
        var prospectImageName = nameComponents[0].trim();

        if (nameComponents.length > 1) {
            var alias = nameComponents[1].trim();

            if (aliasToImageNameMapping[prospectImageName] != undefined) {
                aliasToImageNameMapping[alias] = aliasToImageNameMapping.get(prospectImageName);
            } else {
                aliasToImageNameMapping[alias, prospectImageName];
            }

            baseImage = aliasToImageNameMapping[alias];
        } else {
            baseImage = aliasToImageNameMapping[prospectImageName] != undefined
                ? aliasToImageNameMapping[prospectImageName]
                : prospectImageName;
        }
    }
    return baseImage;
}
var content = `
FROM python:3.9.6-slim
COPY . /app
RUN ./app/run from-csv=/path/to/file.csv
`
console.log(getBaseImageName(content))

// output: -csv=/path/to/file.csv
```

**Comment by DenisRumyantsev** [comment]
@azooinmyluggage, could you please take a look at this ticket since it relates to your area?

**Comment by josheinstein** [comment]
Pretty sure this was already fixed in #15114, but the task running on the hosted agent is still using an old version. I'm not familiar enough with Azure Pipelines to know if I can update this task manually?

**Comment by jonathansp** [comment]
@josheinstein @azooinmyluggage I can confirm #15114 fixed this issue as well.
